File: src/problems/leetcode-cn/283. 移动零/solution/java/Solution.java
class Solution {
    public void moveZeroes(int[] nums) {
        int length = nums.length;
        if (length < 2) return;
        int slow = 0, fast = 0;
        for (; fast < length; fast++) {
            if (nums[fast] != 0) {
                if (fast != slow) {
                    nums[slow] ^= nums[fast];
                    nums[fast] ^= nums[slow];
                    nums[slow] ^= nums[fast];
                }
                slow++;
            }
        }
    }
}

File: src/problems/leetcode-cn/283. 移动零/solution/rust/solution.rs
impl Solution {
    pub fn move_zeroes(nums: &mut Vec<i32>) {
        let length = nums.len();
        if length < 2 {
            return;
        }

        let (mut slow, mut fast) = (0, 0);
        while fast < length {
            if nums[fast] != 0 {
                if fast != slow {
                    (nums[slow], nums[fast]) = (nums[fast], nums[slow]);
                }
                slow += 1;
            }
            fast += 1;
        }
    }
}